autosubmit/monitor/diagram.py

```diff
@@ -162,60 +160,61 @@ def create_stats_report(
-            job_section_output_file = output_file.replace("statistics", "section_summary")
-            headers = JobAggData.headers()
-            _create_table(
-                jobs_data,
-                headers,
-                doc_title=f"SECTION SUMMARY - {expid}",
-                table_title="Aggregated by Job Section"
-            )
-            _create_csv(
-                jobs_data,
-                job_section_output_file,
-                headers
-            )
-            Log.result(f'Section Summary created')
-        if jobs_summary:
-            jobs_data = _get_job_list_data(jobs_list, exp_stats.jobs_stat)
-            jobs_output_file = output_file.replace("statistics", "jobs_summary")
-            headers = JobData.headers()
-            _create_table(
-                jobs_data,
-                headers,
-                doc_title=f"JOBS SUMMARY - {expid}",
-                table_title="Job List"
-            )
-            _create_csv(
-                jobs_data,
-                jobs_output_file,
-                headers
-            )
-            Log.result(f'Jobs Summary created')
-
-        with PdfPages(output_file) as pdf:
-            for figure_number in plt.get_fignums():
-                plt.figure(figure_number)
-                pdf.savefig()
-                plt.close()
-
-            d = pdf.infodict()
-            d['expid'] = expid
-
-        Log.result(f'Stats created at {output_file}')
-
-    return hide
+    if not plot:
+        return False
+
+    if section_summary:
+        jobs_data = _aggregate_jobs_by_section(jobs_list, exp_stats.jobs_stat)
+        job_section_output_file = output_file.replace("statistics", "section_summary")
+        headers = JobAggData.headers()
+        _create_table(
+            jobs_data,
+            headers,
+            doc_title=f"SECTION SUMMARY - {expid}",
+            table_title="Aggregated by Job Section"
+        )
+        _create_csv(
+            jobs_data,
+            job_section_output_file,
+            headers
+        )
+        Log.result(f'Section Summary created')
+    if jobs_summary:
+        jobs_data = _get_job_list_data(jobs_list, exp_stats.jobs_stat)
+        jobs_output_file = output_file.replace("statistics", "jobs_summary")
+        headers = JobData.headers()
+        _create_table(
+            jobs_data,
+            headers,
+            doc_title=f"JOBS SUMMARY - {expid}",
+            table_title="Job List"
+        )
+        _create_csv(
+            jobs_data,
+            jobs_output_file,
+            headers
+        )
+        Log.result(f'Jobs Summary created')
+
+    with PdfPages(output_file) as pdf:
+        for figure_number in plt.get_fignums():
+            plt.figure(figure_number)
+            pdf.savefig()
+            plt.close()
+
+        d = pdf.infodict()
+        d['expid'] = expid
+
+    Log.result(f'Stats created at {output_file}')
+    return True
```

Comment:
The function does the same as before, but if the `plot` is `False` it returns immediately. Otherwise, it creates the report, and returns `True` (to be used to decide whether we have a report to plot or not).